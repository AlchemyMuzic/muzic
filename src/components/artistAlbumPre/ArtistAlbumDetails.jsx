import React from 'react';
import PropTypes from 'prop-types';
import { Link, useParams } from 'react-router-dom';
import imageError from '../imageError/imageError';

export default function ArtistAlbumDetails({ releases }) {
  const { name } = useParams();
  const releasesElements = releases.map(release => (

    <li key={release.release_id}>
      <Link to={`/songs/${release.release_id}/${name}`}>
        <h2>{release.album}</h2>
        <img
          src={release.img}
          alt="release.album"/>
      </Link>
    </li>
  ));

  return (
    <ul data-testid="releaseTest">
      {releasesElements}
    </ul>
  );
}

ArtistAlbumDetails.propTypes = {
  releases: PropTypes.arrayOf(PropTypes.shape({
    release_id: PropTypes.string.isRequired,
    album: PropTypes.string.isRequired
  }))
};
