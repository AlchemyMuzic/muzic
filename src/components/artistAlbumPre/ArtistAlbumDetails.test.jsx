import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import ArtistAlbumDetails from './ArtistAlbumDetails';

describe('ArtistAlbumDetails component', () => {
    afterEach(() => cleanup());
    it('renders a list of releases and their cover art from a selected artist', () => {
        const { asFragment } = render(<BrowserRouter><ArtistAlbumDetails
            releases={[]} /></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    })
})
