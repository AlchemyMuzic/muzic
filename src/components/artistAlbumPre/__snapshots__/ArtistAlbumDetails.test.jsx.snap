// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistAlbumDetails component renders a list of releases and their cover art from a selected artist 1`] = `
<DocumentFragment>
  <ul
    data-testid="releaseTest"
  />
</DocumentFragment>
`;
